### PR TITLE
driver: Refactor Landing_pages to use Format

### PR DESCRIPTION
**Pull request by Julow**
Util.write_file is refactored to be more suitable.

### CODE REVIEW

File: src/driver/util.ml [hunk 1]
```diff
@@ -17,29 +18,18 @@ let lines_of_process cmd =
...
+let with_out_to filename f =
+  let open ResultMonad in
+  OS.Dir.create (Fpath.parent filename) >>= fun _ ->
+  OS.File.with_oc filename
+    (fun oc () ->
+      f oc;
+      Ok ())
+    ()
+  >>= fun r ->
+  Result.get_ok r;
```
Comment (panglesd) [inline src/driver/util.ml]: Since we are in the result monad, why not avoid calling `get_ok` ?
```
|> Result.join >>= fun () ->
```
seems better to me!

Comment (panglesd) [reply src/driver/util.ml]: I realize that with the amount of `get_ok` in this driver, that suggestion does not make a lot of sense!
But still :)

**Review by panglesd — APPROVED** [review]
I like most changes!
I don't find that the `@\n` improves readability... But I'm ok with it!

**Comment by panglesd** [comment]
Let's go!

File: src/driver/util.ml (hunk 1 above)
Comment (Julow) [reply src/driver/util.ml]: The idea was that `r` is always `Ok ()` and so I use the exception function. `Result.join` works too!

**Comment by Julow** [comment]
Format has a more complex syntax than string literals but I think getting rid of ` |> List.map s_of_pkg |> String.concat "\n"` is worth it.

**Pull request merged**